### DISCUSSION TITLE
Solving classic quadratic equations

### DISCUSSION THREAD

**Discussion by nicola-gigante** [cvc5/cvc5#10984] [Q&A] [answered]
I'm preparing a set of simple SMT examples so I wrote this file:

```
(set-logic ALL)
(set-option :produce-models true)

(declare-const x Real)
(define-const a Real 1.0)
(define-const b Real (- 8.0))
(define-const c Real 16.0)

(assert (= (+ (* a x x) (* b x) c) 0.0))

(check-sat)
(get-model)
```

This is a classic quadratic equation $x^2 - 8x + 16 = 0$, as those thought in school, which has a unique solution $x = 4$ because the polynomial is a square.

Z3 says it is satisfiable and shows $x = 4$ indeed.
`cvc5` hangs instead, with the following command line:
```
cvc5 --lang=smtlib2 test.smt
```

Is there some option I have to pass to `cvc5` to make it able to solve these kinds of problems?

**Comment by ajreynol** [comment, marked as answer]
You can solve this problem using the option `--nl-cov`, otherwise setting the logic to `QF_NRA` also will solve this problem since it implies that option is set to true as well.

**Reply by nicola-gigante** [reply]
Thanks, that worked!
